### PR TITLE
Add Data provider to Insights card

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
#### Summary of Changes (What does this PR do?)

- [x] Add Data provider to Insights card

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks @BenjaminSsempala for this.
Will this PR require the corresponding backend PR to first get merged or things can move separately?

**Comment by BenjaminSsempala** [comment]
> Thanks @BenjaminSsempala for this. Will this PR require the corresponding backend PR to first get merged or things can move separately?

Yes @Baalmart Preferably the backend PR should be merged first so that I can effectively test out . Thanks

**Comment by Baalmart** [comment]
Okay @BenjaminSsempala I must have left a small comment on the backend PR. Did you see it?

File: mobile/lib/models/air_quality_reading.g.dart [hunk 1]
```diff
@@ -114,7 +114,7 @@ Site _$SiteFromJson(Map<String, dynamic> json) {
     searchLocation: json['location_name'] as String? ?? '',
     country: json['country'] as String? ?? '',
     region: json['region'] as String? ?? '',
-    source: json['network'] as String? ?? '',
+    dataProvider: json['network'] as String? ?? '',
```
Comment (NoahNsimbe) [inline mobile/lib/models/air_quality_reading.g.dart:117]: Thanks @BenjaminSsempala, did you make these changes yourself, or they were made using build runner??..

**Comment by BenjaminSsempala** [comment]
> Okay @BenjaminSsempala I must have left a small comment on the backend PR. Did you see it?

Thanks @Baalmart I've replied to the comment on that PR

File: mobile/lib/models/air_quality_reading.g.dart (hunk 1 above)
Comment (BenjaminSsempala) [reply mobile/lib/models/air_quality_reading.g.dart]: Thanks @NoahNsimbe
I've regenerated this file using build runner

**Review by NoahNsimbe — APPROVED** [review]
LGTM Thanks @BenjaminSsempala

**Pull request merged**